airflow/executors/executor_loader.py

```diff
@@ -202,10 +202,10 @@ def lookup_executor_name_by_str(cls, executor_name_str: str) -> ExecutorName:
         elif executor_name := _module_to_executors.get(executor_name_str):
             return executor_name
         else:
-            raise AirflowException(f"Unknown executor being loaded: {executor_name}")
+            raise AirflowException(f"Unknown executor being loaded: {executor_name_str}")
 
     @classmethod
-    def load_executor(cls, executor_name: ExecutorName | str) -> BaseExecutor:
+    def load_executor(cls, executor_name: ExecutorName | str | None) -> BaseExecutor:
```

Comment:
We now except None (which means In some cases that there was no specified executor, so in that case we should return the default executor).